src/Psalm/Internal/Type/TemplateInferredTypeReplacer.php

```diff
@@ -451,7 +451,10 @@ private static function replaceConditional(
                     null,
                     false,
                     false,
-                )) {
+                )
```

Comment:
The logic is

When doing `A is B ? C : D`:
- If A is contained by B => It's C
- If B is contained by A => It's D
- Else, it's C|D

I think this logic is wrong and does not work well with string type like `literal-string` and `non-empty-string`.
it should be
- If A is contained by B => It's C
- If A&B = NEVER => It's D
- Else it's C|D

But I dunno if a check `A&B=Never` already exist for psalm, so I added a special check for string instead.